src/commonMain/kotlin/fr/acinq/lightning/payment/IncomingPaymentHandler.kt

```diff
@@ -243,82 +219,192 @@ class IncomingPaymentHandler(val nodeParams: NodeParams, val db: IncomingPayment
-                            if (payment.parts.filterIsInstance<PayToOpenPart>().isNotEmpty()) {
-                                // We consider the total amount received (not only the pay-to-open parts) to evaluate whether or not to accept the payment
-                                val payToOpenFee = payment.parts.filterIsInstance<PayToOpenPart>().map { it.payToOpenRequest.payToOpenFeeSatoshis }.sum()
-                                nodeParams.liquidityPolicy.value.maybeReject(payment.amountReceived, payToOpenFee.toMilliSatoshi(), LiquidityEvents.Source.OffChainPayment, logger)?.let { rejected ->
-                                    logger.info { "rejecting pay-to-open: reason=${rejected.reason}" }
-                                    nodeParams._nodeEvents.emit(rejected)
-                                    val actions = payment.parts.map { part ->
-                                        val failureMsg = TemporaryNodeFailure
-                                        when (part) {
-                                            is HtlcPart -> actionForFailureMessage(failureMsg, part.htlc)
-                                            is PayToOpenPart -> actionForPayToOpenFailure(privateKey, failureMsg, part.payToOpenRequest) // NB: this will fail all parts, we could only return one
+                            val htlcParts = payment.parts.filterIsInstance<HtlcPart>()
+                            val willAddHtlcParts = payment.parts.filterIsInstance<WillAddHtlcPart>()
+                            when {
+                                willAddHtlcParts.isNotEmpty() -> when (val result = validateOnTheFlyFundingRate(willAddHtlcParts.map { it.amount }.sum(), currentFeerate)) {
+                                    is Either.Left -> rejectOnTheFlyFunding(payment, incomingPayment, result.value)
+                                    is Either.Right -> {
+                                        val (requestedAmount, fundingRate) = result.value
+                                        val addToFeeCredit = run {
+                                            val featureOk = nodeParams.features.hasFeature(Feature.FundingFeeCredit)
+                                            val remoteNodeOk = remoteFundingRates.paymentTypes.contains(LiquidityAds.PaymentType.FromFeeCredit)
+                                            // We use an arbitrary threshold that is higher than just the current liquidity fees.
+                                            // This reduces the frequency of on-chain operations for payments that are about the size of the fees.
+                                            // It also ensures that if we end up splicing with a higher feerate because we have unconfirmed parent
+                                            // transactions, we will have enough fee credit to cover this higher feerate.
+                                            val feeCreditThreshold = fundingRate.fees(currentFeerate, requestedAmount, requestedAmount).total * 5
```

Comment:
The 5x multiplier is maybe too high and can end up with a non-negligible fee credit that stays forever on the LSP side. Should we reduce this to a simple 2x, with the risk that it creates more frequent on-chain transactions?